ui/tests/acceptance/auth-test.js

```diff
@@ -62,62 +62,112 @@ module('Acceptance | auth', function (hooks) {
+          payload: {
+            redirect_uri: 'http://localhost:7357/ui/vault/auth/custom-jwt/oidc/callback',
+            role: 'some-role',
+          },
         },
         oidc: {
-          included: 'role',
-          url: '/v1/auth/oidc/oidc/auth_url',
+          url: '/v1/auth/custom-oidc/oidc/auth_url',
+          payload: {
+            redirect_uri: 'http://localhost:7357/ui/vault/auth/custom-oidc/oidc/callback',
+            role: 'some-role',
+          },
         },
         radius: {
-          included: 'password',
-          url: '/v1/auth/radius/login/null',
+          url: '/v1/auth/custom-radius/login/some-username',
+          payload: {
+            password: 'some-password',
+          },
         },
         github: {
-          included: 'token',
-          url: '/v1/auth/github/login',
+          url: '/v1/auth/custom-github/login',
+          payload: {
+            token: 'some-token',
+          },
+        },
+        saml: {
+          url: '/v1/auth/custom-saml/sso_service_url',
+          payload: {
+            role: 'some-role',
+          },
         },
       };
     });
 
-    for (const backend of supportedAuthBackends().reverse()) {
-      test(`for ${backend.type}`, async function (assert) {
+    for (const backend of allSupportedAuthBackends().reverse()) {
+      test(`for ${backend.type} ${
+        ENT_AUTH_METHODS.includes(backend.type) ? '(enterprise)' : ''
+      }`, async function (assert) {
         const { type } = backend;
-        const isOidc = ['jwt', 'oidc'].includes(type);
-        // OIDC types make 3 requests, each time the role changes
+        const expected = this.expected[type];
+        const isOidc = ['oidc', 'jwt'].includes(type);
         assert.expect(isOidc ? 6 : 2);
+
         this.assertReq = (req) => {
           const body = type === 'token' ? req.requestHeaders : JSON.parse(req.requestBody);
-          const { included, url } = this.expected[type];
-          assert.true(Object.keys(body).includes(included), `${type} includes ${included}`);
-          assert.strictEqual(req.url, url, `${type} calls the correct URL`);
+          if (isOidc && !body.role) {
+            // OIDC and JWT auth form calls the endpoint every time the role or mount is updated.
+            // if role is not provided, it means we haven't filled out the full info yet so don't
+            // validate the payload until all data is provided
+            // eslint-disable-next-line qunit/no-early-return
+            return {};
+          }
+          assert.strictEqual(req.url, expected.url, `${type} calls the correct URL`);
+          Object.keys(expected.payload).forEach((expKey) => {
+            assert.strictEqual(
+              body[expKey],
+              expected.payload[expKey],
+              `${type} payload includes ${expKey} with expected value`
+            );
+          });
         };
         await visit('/vault/auth');
         await component.selectMethod(type);
```

Comment:
should we update these selectors as we can so we can slowly move away from `ember-cli-page-object`?

Reply:
Since we're not backporting this, I think that's a good idea!